[ao] Updated ModelReport function calls to show not dependent on Fx GraphMode

vspenubarthi opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #81369
* #81382
* __->__ #81252
* #81251
* #81250
* #81249
* #81174

Summary: Before, all the function calls for the ModelReport object were
dependent on the Fx Graph Mode workflow. However, in reality, this was
not true and the only requirement that was needed was for the model to
be a traceable GraphModule. This also helped keep the ModelReport class
as detached from the Fx Workflow as possible so that it can be used as a
more all purpose tool in the future.

This updated all the references to make sure that it wasn't specifically
referencing that a Fx Graph Mode workflow is needed, and is instead more
general since all we really need is a traceable model.

Test Plan: python test/test_quantization.py TestFxModelReportClass

Reviewers:

Subscribers:

Tasks:

Tags:

vspenubarthi committed:
[ao] Updated ModelReport function calls to show not dependent on Fx GraphMode

Summary: Before, all the function calls for the ModelReport object were
dependent on the Fx Graph Mode workflow. However, in reality, this was
not true and the only requirement that was needed was for the model to
be a traceable GraphModule. This also helped keep the ModelReport class
as detached from the Fx Workflow as possible so that it can be used as a
more all purpose tool in the future.

This updated all the references to make sure that it wasn't specifically
referencing that a Fx Graph Mode workflow is needed, and is instead more
general since all we really need is a traceable model.

Test Plan: python test/test_quantization.py TestFxModelReportClass

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vspenubarthi committed:
Update on "[ao] Updated ModelReport function calls to show not dependent on Fx GraphMode"


Summary: Before, all the function calls for the ModelReport object were
dependent on the Fx Graph Mode workflow. However, in reality, this was
not true and the only requirement that was needed was for the model to
be a traceable GraphModule. This also helped keep the ModelReport class
as detached from the Fx Workflow as possible so that it can be used as a
more all purpose tool in the future.

This updated all the references to make sure that it wasn't specifically
referencing that a Fx Graph Mode workflow is needed, and is instead more
general since all we really need is a traceable model.

Test Plan: python test/test_quantization.py TestFxModelReportClass

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vspenubarthi committed:
Update on "[ao] Updated ModelReport function calls to show not dependent on Fx GraphMode"


Summary: Before, all the function calls for the ModelReport object were
dependent on the Fx Graph Mode workflow. However, in reality, this was
not true and the only requirement that was needed was for the model to
be a traceable GraphModule. This also helped keep the ModelReport class
as detached from the Fx Workflow as possible so that it can be used as a
more all purpose tool in the future.

This updated all the references to make sure that it wasn't specifically
referencing that a Fx Graph Mode workflow is needed, and is instead more
general since all we really need is a traceable model.

Test Plan: python test/test_quantization.py TestFxModelReportClass

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vspenubarthi committed:
Update on "[ao] Updated ModelReport function calls to show not dependent on Fx GraphMode"


Summary: Before, all the function calls for the ModelReport object were
dependent on the Fx Graph Mode workflow. However, in reality, this was
not true and the only requirement that was needed was for the model to
be a traceable GraphModule. This also helped keep the ModelReport class
as detached from the Fx Workflow as possible so that it can be used as a
more all purpose tool in the future.

This updated all the references to make sure that it wasn't specifically
referencing that a Fx Graph Mode workflow is needed, and is instead more
general since all we really need is a traceable model.

Test Plan: python test/test_quantization.py TestFxModelReportClass

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on torch/ao/quantization/fx/_model_report/model_report.py:
```diff
@@ -111,9 +111,9 @@ def get_observers_of_interest(self) -> Dict[str, Set[str]]:
         return self._detector_name_to_observer_fqns.copy()
 
-    def prepare_detailed_calibration(self, prepared_fx_model: GraphModule) -> GraphModule:
+    def prepare_detailed_calibration(self, model: GraphModule) -> GraphModule:
```
this still assumes the model is symbolically traced, I think ideally we should update the type to `nn.Module` for top level apis and do tracing inside the function

vspenubarthi replied:
I think we can have a further discussion about this, but i feel like doing this would make it harder to integrate this into quantization workflow. Specifically, how would we handle the flow:

Initialize model &rarr; Prepare model &rarr; Callibrate model &rarr; Convert model &rarr; ...

that we typically find in quantization. If we were to do the tracing inside the function, then that conflicts with what the `prepare` step is doing in the process.

So I think i am still confused how / why we would want to do tracing within ModelReport

jerryzh168 replied:
I'm not sure why do we have to use this inside quantization flow, this sounds something that is independent of quantization flow itself, maybe we can align on that first

vspenubarthi committed:
Update on "[ao] Updated ModelReport function calls to show not dependent on Fx GraphMode"


Summary: Before, all the function calls for the ModelReport object were
dependent on the Fx Graph Mode workflow. However, in reality, this was
not true and the only requirement that was needed was for the model to
be a traceable GraphModule. This also helped keep the ModelReport class
as detached from the Fx Workflow as possible so that it can be used as a
more all purpose tool in the future.

This updated all the references to make sure that it wasn't specifically
referencing that a Fx Graph Mode workflow is needed, and is instead more
general since all we really need is a traceable model.

Test Plan: python test/test_quantization.py TestFxModelReportClass

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vspenubarthi committed:
Update on "[ao] Updated ModelReport function calls to show not dependent on Fx GraphMode"


Summary: Before, all the function calls for the ModelReport object were
dependent on the Fx Graph Mode workflow. However, in reality, this was
not true and the only requirement that was needed was for the model to
be a traceable GraphModule. This also helped keep the ModelReport class
as detached from the Fx Workflow as possible so that it can be used as a
more all purpose tool in the future.

This updated all the references to make sure that it wasn't specifically
referencing that a Fx Graph Mode workflow is needed, and is instead more
general since all we really need is a traceable model.

Test Plan: python test/test_quantization.py TestFxModelReportClass

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vspenubarthi commented:
@pytorchbot merge -g

vspenubarthi commented:
@pytorchbot merge -g
